libs/designsystem/src/lib/components/chart-wip/chart-js/chart-js.service.ts

```diff
@@ -0,0 +1,232 @@
+  }
+
+  private initializeNewChart(canvasElement: HTMLCanvasElement, config: ChartConfiguration) {
+    this.chart = new Chart(canvasElement, config);
+  }
+
+  private getTypeConfig(type: ChartType) {
+    /* Deep copy to avoid Chart object modifying parts of CHART_TYPE_CONFIGS 
+    as it copies by reference when initialized */
+    return deepCopy(CHART_TYPE_CONFIGS[type]);
+  }
+
+  private createBlankLabels(datasets: ChartDataset[]): string[] {
+    const largestDataset = datasets.reduce((previousDataset, currentDataset) =>
+      previousDataset.data.length > currentDataset.data.length ? previousDataset : currentDataset
+    );
+    return Array(largestDataset.data.length).fill('');
+  }
+
+  private getAnnotationDefaults(type: string) {
+    return CHART_ANNOTATION_CONFIGS[type];
+  }
+
+  private applyDefaultsToAnnotations(annotations: AnnotationOptions[]) {
+    return annotations.map((annotation) => {
+      const annotationTypeDefaults = this.getAnnotationDefaults(annotation.type);
+      return deepMergeObjects(annotationTypeDefaults, annotation);
+    });
+  }
+
+  private createAnnotationPluginOptionsObject(annotations: AnnotationOptions[]) {
+    const annotationsWithDefaults = this.applyDefaultsToAnnotations(annotations);
+    return {
+      plugins: {
+        annotation: {
+          annotations: annotationsWithDefaults,
+        },
+      },
+    };
+  }
+
+  private applyInteractionFunctionsExtensions(options: ChartOptions): ChartOptions {
+    const interactionFunctionsExtensions = INTERACTION_FUNCTIONS_EXTENSIONS;
+    Object.entries(interactionFunctionsExtensions).forEach(([key, _]) => {
+      const callback = options[key];
+      options[key] = (e: Event, a: ActiveElement[], c: Chart) => {
+        interactionFunctionsExtensions[key](e, a, c, callback);
+      };
+    });
+    return options;
+  }
+
+  private createOptionsObject(args: {
+    type: ChartType;
+    customOptions?: ChartOptions;
+    annotations?: AnnotationOptions[];
+  }): ChartOptions {
+    const { type, customOptions, annotations } = args;
+
+    const typeConfig = this.getTypeConfig(type);
+    const typeConfigOptions = typeConfig?.options;
+    const annotationPluginOptions = annotations
+      ? this.createAnnotationPluginOptionsObject(annotations)
+      : {};
+    let options: ChartOptions = deepMergeObjects(
+      typeConfigOptions,
+      customOptions,
+      annotationPluginOptions
+    );
+    return this.applyInteractionFunctionsExtensions(options);
+  }
+
+  private createConfigurationObject(
+    type: ChartType,
+    datasets: ChartDataset[],
+    options: ChartOptions,
+    dataLabels?: unknown[]
+  ): ChartConfiguration {
+    /* chartJS requires labels; if none is provided create an empty string array
+    to make it optional for consumer */
+    const labels = !dataLabels ? this.createBlankLabels(datasets) : dataLabels;
+    const typeConfig = this.getTypeConfig(type);
+    //TODO: do we need deep merge here?
```

Comment:
Remember this todo. 

My take is that it can't hurt (famous last words), and then it would be nicely aligned with `createOptionsObject()`

Reply:
I opted for deep merge :D